smapply/api.py

```diff
@@ -0,0 +1,323 @@
+        user = process_user(serializer.data)
+    else:
+        user = profile.user
+    if webhook.award_id is not None:
+        klass_info = SMApplyAPI().get('/awards/{}'.format(webhook.award_id)).json()
+        personal_price = klass_info.value
+
+        klass = Klass.objects.filter(klass_key=webhook.award_id, source=ApplicationSource.SMAPPLY).first()
+        if not klass:
+            if not personal_price:
+                raise SMApplyException(
+                    "Klass has no price and klass_key %s does not exist",
+                    webhook.award_id
+                )
+            bootcamp = Bootcamp.objects.create(title=klass_info['name'])
+            klass = Klass.objects.create(
+                bootcamp=bootcamp,
+                title=klass_info['description'],
+                klass_key=klass_info['id'])
+            try:
+                MailgunClient().send_individual_email(
+                    "Klass and Bootcamp created, for klass_key {klass_key}".format(
+                        klass_key=klass_info['id']
+                    ),
+                    "Klass and Bootcamp created, for klass_key {klass_key}".format(
+                        klass_key=klass_info['id']
+                    ),
+                    settings.EMAIL_SUPPORT
+                )
+            except:  # pylint: disable=bare-except
+                log.exception(
+                    "Error occurred when sending the email to notify "
+                    "about Klass and Bootcamp creation for klass key %s",
+                    klass_info['id']
+                )
+        if personal_price is not None:
+            user.klass_prices.update_or_create(
+                klass=klass,
+                defaults={'price': personal_price}
+            )
+        else:
+            user.klass_prices.filter(klass=klass).delete()
+
+
+def list_users():
+    """
+    Generator for all users in SMApply
+
+    Yields:
+        dict: SMApply user as dict
+
+    """
+    smapi = SMApplyAPI()
+    url = 'users'
+    while url:
+        response = smapi.get(url).json()
+        users = response['results']
+        for sma_user in users:
+            yield sma_user
+        next_page = urlparse(response['next']).query
+        url = 'users?{}'.format(next_page) if next_page else None
+
+
+def post_payment(order):
+    """
+    Update amount paid by a user for a class when an order is fulfilled.
+
+    Args:
+        order(Order): the Order object to send payment info about to SMApply
+
+    """
+    if order.status != Order.FULFILLED:
+        return
+    klass = order.get_klass()
+    user = order.user
+    if not klass or klass.bootcamp.legacy:
+        return
+    total_paid = Line.total_paid_for_klass(order.user, klass.klass_key).get('total') or Decimal('0.00')
+    payment_metadata = {
+        'value': '{:0.2f}'.format(total_paid)
+    }
+    webhook = WebhookRequestSMA.objects.filter(user_id=user.profile.fluidreview_id, award_id=klass.klass_key).last()
```

Comment:
I think this should be `user.profile.smapply_id` instead of `user.profile.fluidreview_id`